### ISSUE TITLE
Create a stack in the UI with invalid yaml or errors in the services and stuck in Activating

### ISSUE THREAD

**Issue by ibuildthecloud** [rancher/rancher#3950]

**Comment by Rucknar** [comment]
I've seen this a few times also. Occasionally you can see an error around the yaml syntax when your launching the stack but not always. 

We've ended up adding a validate script we run before initiating compose that validates the yaml file using a ruby gem.

**Comment by deniseschannon** [comment]
Confirmed this is working but causing some issues with catalog. 

https://github.com/rancher/rancher/issues/4059

Closing in favor of the other one.

**Comment by deniseschannon** [comment]
We had to revert the change for GA, but will put it in the next release of rancher-compose.

**Comment by deniseschannon** [comment]
When launching stacks in the UI, they are getting stuck in activating. We need to update the stack state to "Error" when there is an error in rancher-compose. 

@joshwget Can you investigate if this is a rancher-compose issue or if we need a cattle change as well?

**Comment by deniseschannon** [comment]
Currently, with rancher-compose, we immediately throw an error when there is an invalid yaml. Cattle needs to be able to handle this error.

**Comment by alena1108** [comment]
@deniseschannon the fix is on rancher-compose side

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.2.0-pre4-rc1.

Scenario 1:
When launching stacks in the UI with invalid yml files, user is presented with the error message about the error like  ```Error (Unsupported config option for test1 service: 'label')````  and stack is in "error" state.

We are not able to create new services under this stack .User is presented with error message - ```Validation failed in API: stackId```

Scenario 2:
When launching stacks in the UI with invalid yml for 1 of the services ( and valid yml for other services) , user is presented with the error message about the error like  ```Error (Unsupported config 
option for test1 service: 'label')````  for the service with invalid yml. Stack is in "error" state.
Services with valid yml get created in "inactive" state. Should we not even be creating this service ?

**Comment by alena1108** [comment]
Rancher-compose should validate ALL services from the compose file before starting their creation.

**Comment by joshwget** [comment]
@sangeethah @alena1108 There are actually two cases under Scenario 2.

1. There's an error in the Compose file itself, such as a syntax error or a wrong type used somewhere. Regardless of where this occurs in the file, the stack will throw an error and no services will be created.
2. There's a semantic error in one of the attributes for a service. In this case Rancher Compose can't validate everything ahead of time since it's creating services one at a time.

Case 1 seems to be fine. To fix case 2 I'd need some sort of validation API for a launch config. In any case, this feels a little separate from the original issue which looks to be fixed by the PR from @alena1108.

**Comment by deniseschannon** [comment]
@sangeethah I'm going to close this as it addresses the original issue. Can you open a new issue for the 2nd use case that Josh mentioned?
